Sources/TokamakCore/Environment/EnvironmentKey.swift

```diff
@@ -14,19 +14,29 @@
 
 public protocol EnvironmentKey {
   associatedtype Value
-  static var defaultValue: Self.Value { get }
+  static var defaultValue: Value { get }
 }
 
-public struct _EnvironmentKeyWritingModifier<Value>: ViewModifier {
-  public typealias Body = Never
+protocol EnvironmentModifier {
+  func _modifyEnvironment(_ values: inout EnvironmentValues)
```

Comment:
**suggestion:** This is a real `internal` function, so no need to underscore it. I think at this point the rules should look something like this:

1. If a function or a type are restricted to a module and has no `public` access control, no need for an underscore then.
2. If a function or a type are part of a public renderer module API (e.g. `TokamakDOM`), no need for an underscore, users may use those symbols directly, and parts of `TokamakCore` would be re-exported by the renderer module via `public typealias`.
3. If a function or a type have `public` on them only by necessity to make them available in `TokamakDOM` but unavailable to users (or not intended for public use), underscore is needed to indicate that.

This is all caused by the lack of "package private" access control in Swift. And we do need separate modules to allow for separate renderers for different platforms. I want to add a separate `TokamakHTML` with a renderer for static websites soon, and then I'd like to try to implement a native renderer for Android, probably in a separate `TokamakAndroid` module. The benefit of separate modules here is that users can pick and choose what they want, i.e. purely static websites go with `TokamakHTML`, single-page apps would use `TokamakDOM`, maybe in conjuction with `TokamakHTML` for pre-rendering, Android apps would use `TokamakAndroid` with a need to be aware of any of the web modules.

CC @j-f1, I'm going to add this to `README.md` to clarify it.

```suggestion
  func modifyEnvironment(_ values: inout EnvironmentValues)
```